extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/authentication/validator/LDAPCredentialsValidator.java

```diff
@@ -213,8 +221,10 @@ SearchResult getLdapUserObject(BasicAuthLDAPConfig ldapConfig, DirContext contex
   boolean validatePassword(BasicAuthLDAPConfig ldapConfig, LdapName userDn, char[] password)
   {
     InitialDirContext context = null;
+    ClassLoader currentClassLoader = Thread.currentThread().getContextClassLoader();
```

Comment:
nit: Can't we remove setting the classloader here ? 
Push the finally block after `validatePassword` and change the `validatePassword` scope to private ?

Reply:
The thought did cross my mind. I didn't want to tinker with the try/catch flow. I also don't know what the impact of opening `InitialDirContext` while one is already open, will be. Probably none. This change felt safer to do.